circuit/src/ecc/generic/mod.rs

```diff
@@ -1,2 +1,47 @@
-/// Native implementation
-pub mod native;
+pub struct EccAddChipset<C: CurveAffine, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P>
+where
+    P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS>,
+    C::Base: FieldExt,
+    C::Scalar: FieldExt,
+{
+    p: AssignedPoint<C, N, NUM_LIMBS, NUM_BITS, P>,
+    q: AssignedPoint<C, N, NUM_LIMBS, NUM_BITS, P>,
+}
+impl<C: CurveAffine, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P> Chipset for EccAddChipset<C, N, NUM_LIMBS, NUM_BITS, P>
+where
+    P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS>,
+    C::Base: FieldExt,
+    C::Scalar: FieldExt,
+{
+    type Config = EccAddConfig;
+    type Output = AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>;
+    fn synthesize(
+        self, common: &CommonConfig, config: &Self::Config, layouter: impl Layouter<F>,
+    ) -> Result<Self::Output, Error> {
+        // Implement the synthesis method
+        ...
```

Comment:
This should be implemented correctly. The logic is the same as the one for `EccAddChipset ` inside `circuit/src/ecc/same_curve/mod.rs`